test/lib/builder/builder.js

```diff
@@ -105,6 +105,35 @@ test("Build application.a [dev mode]", (t) => {
 	});
 });
 
+test("Build application.a and clear target path [dev mode]", (t) => {
```

Comment:
This test does execute the newly added lines, but doesn't really test whether something got removed.

I would rather mock `rimraf` for this test and check whether it got called with the correct argument.  But we can add that test in a follow up change from our side, so feel free to leave this one as is.

**Example for mock-testing:**
https://github.com/SAP/ui5-builder/blob/3bf0f56097163ad04b3dfd1e9efab40ccd3607b5/test/lib/types/library/LibraryFormatter.js#L168-L174

Reply:
Yes, it does test whether the folder has been deleted. Build is called twice. The first one creates a sub folder called "/rubbish-should-be-deleted" and clearDest=false. The second, builds with clearDest=true.

If you call the second build with clearDest=false, test will fail due to **assert.directoryDeepEqual(destPath, expectedPath)**

![test](https://user-images.githubusercontent.com/443888/55268303-8807b500-52dc-11e9-9841-34eac902c6b3.PNG)

Reply:
I see, now I get it 👍